File: src/main/java/ru/ioleksiv/telegram/bot/api/annotations/filter/secondary/MaskPositionFilter.java
package ru.ioleksiv.telegram.bot.api.annotations.filter.secondary;

import ru.ioleksiv.telegram.bot.api.annotations.filter.primitive.StringFilter;
import ru.ioleksiv.telegram.bot.api.model.annotation.AnnotationState;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.METHOD)
@Documented
public @interface MaskPositionFilter {

    AnnotationState value() default AnnotationState.ON;

    StringFilter point() default @StringFilter(AnnotationState.OFF);

}

File: core/src/main/kotlin/io/github/oleksivio/tl/kbot/core/annotations/filter/composite/MaskPositionFilter.kt
package io.github.oleksivio.tl.kbot.core.annotations.filter.composite

import io.github.oleksivio.tl.kbot.core.annotations.filter.primitive.FloatFilter
import io.github.oleksivio.tl.kbot.core.model.annotation.AnnotationState
import io.github.oleksivio.tl.kbot.server.api.objects.std.sticker.MaskPosition

@Retention
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@MustBeDocumented
annotation class MaskPositionFilter(
    val status: AnnotationState = AnnotationState.ON,
    val validator: Array<String> = [],
    val point: MaskPosition.Type = MaskPosition.Type.UNKNOWN,
    val xShift: FloatFilter = FloatFilter(
        status = AnnotationState.OFF
    ),
    val yShift: FloatFilter = FloatFilter(
        status = AnnotationState.OFF
    ),
    val scale: FloatFilter = FloatFilter(
        status = AnnotationState.OFF
    )
)
